### DISCUSSION TITLE
build pg_duckdb in docker got error.

### DISCUSSION THREAD

**Discussion by lucasdengcn** [duckdb/pg_duckdb#239] [Q&A] [answered]
I tried to build **pg_duckdb** for **PostgreSQL 16** in **debian bullseye**.

### command to build pg_duckdb

git clone --recurse-submodules https://github.com/duckdb/pg_duckdb; 
cd pg_duckdb; 
make install;

### error as following, detail as attached.

src/pgduckdb_filter.cpp:33:30: error: no member named 'string_view' in namespace 'std'

[error.log](https://github.com/user-attachments/files/17199462/error.log)

Is there anything i missed ?

**Comment by Tishj** [comment]
I assume your environment does not have c++ >= 17 installed, as that is when `string_view` was introduced

**Comment by JelteF** [comment]
Yeah, it indeed looks like there's no C++17 available.  I'd expect compilation to fail in a different way though in that case, because we provide the `-std=c++17` flag in our Makefile.

**Comment by lucasdengcn** [comment]
@JelteF @Tishj , I have added libstdc++-10-dev. but still the same error.

**Comment by wuputah** [comment]
I know libstdc++-12-dev works, as it's on my branch here: https://github.com/duckdb/pg_duckdb/blob/jd/docker-image/Dockerfile

**Comment by wuputah** [comment, marked as answer]
@lucasdengcn give it a try now that #248 is merged

**Comment by lucasdengcn** [comment]
@wuputah thank you. it works.
